src/settings/settings.service.ts

```diff
@@ -49,4 +49,44 @@ export async function updateProfileInfo(
 		console.error('Failed to update profile information', err, { profile, variables });
 		throw err;
 	}
-}
+};
+
+export const fetchNewsletterPreferences = async (email: string) => {
+	try {
+		const response = await fetch(
+			`${getEnv('PROXY_URL')}/campaign-monitor/preferences?email=${email}`,
+			{
+				method: 'GET',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				credentials: 'include',
+			}
+		);
+
+		return response.json();
```

Comment:
add check before call to .json() if the status is between [200, 400[

![image](https://user-images.githubusercontent.com/1710840/81386488-58e58900-9115-11ea-8ed8-02333e3da901.png)